subsys/mgmt/mcumgr/transport/Kconfig.lorawan

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (c) 2024, Jamie McCrae
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# The Kconfig file is dedicated to the LoRaWAN transport of MCUmgr
+# subsystem and provides Kconfig options to control aspects of
+# the transport.
+#
+# Options defined in this file should be prefixed:
+#  MCUMGR_TRANSPORT_LORAWAN_
+
+menuconfig MCUMGR_TRANSPORT_LORAWAN
+	bool "LoRaWAN MCUmgr SMP transport"
+	depends on LORAWAN
+	help
+	  Enables handling of SMP commands received over LoRa.
+
+if MCUMGR_TRANSPORT_LORAWAN
+
+config MCUMGR_TRANSPORT_LORAWAN_FRAME_PORT
+	int "LoRaWAN SMP frame port"
+	range 0 120
+	default 2
+	help
+	  LoRaWAN download and uplink frame port used for communication. All messages received on
+	  this port will be treated as SMP packets.
+
+config MCUMGR_TRANSPORT_LORAWAN_CONFIRMED_PACKETS
+	bool "Use confirmed packets for uplinks"
+	default y
+	help
+	  Will use confirmed uplink packets for responses if enabled, otherwise will use
+	  unconfirmed packets.
+
+config MCUMGR_TRANSPORT_LORAWAN_REASSEMBLY
+	bool "Reassemble LoRaWAN SMP messages"
+	select MCUMGR_TRANSPORT_REASSEMBLY
+	default y
+	help
+	  Will reassemble downlink LoRaWAN messages together to allow for messages larger than a
+	  single message to be received, otherwise will support messages up to a single packet in
+	  size.
+
+menuconfig MCUMGR_TRANSPORT_LORAWAN_POLL_FOR_DATA
+	bool "Send empty packet if partial packet received"
+	depends on MCUMGR_TRANSPORT_LORAWAN_REASSEMBLY
+	default y
+	help
+	  Will send an empty packet if a partial (fragmented) message has been received from the
+	  server, this will allow the next packet to be received without waiting for next
+	  transmission window.
+
+	  Note: this requires a dedicated thread in order to prevent blocking the system workqueue.
```

Comment:
There is an `FPending` flag for downlink messages for exactly this purpose. Can we not use that instead?

Quote from the spec:

**4.3.1.4 Frame pending bit (FPending in FCtrl, downlink only)**
The frame pending bit (FPending) is only used in downlink communication, indicating that
the network has more data pending to be sent and therefore asking the end-device to open
another receive window as soon as possible by sending another uplink message.

Reply:
I don't see how that would work any differently? If the flag is set, you still all the same code, the way it is right now means it only sends an empty frame if there is outstanding data for the current MCUmgr packet which in my mind is all this should handle, something non-MCUmgr can be retrieved by the application code

Reply:
I thought the LoRaWAN stack would handle the FPending flag automatically, but it doesn't seem to be the case. I agree your implementation makes sense.